internal/thelma/cli/execution.go

```diff
@@ -44,6 +45,7 @@ func newExecution(options *executionOptions, leafNode *node, builder builder.The
 // all errors are aggregated into a RunError
 func (e *execution) execute() error {
 	e.startTime = time.Now()
+	e.setFlagsFromEnvironment()
```

Comment:
this approach looks reasonable. I think it would be fine to to just return the error without making this a hook (conceptually), but maybe I'm missing some subtlety in how the existing CLI code works... Apologies for the lack of unit test coverage here.

Reply:
Honestly it's that I wanted the cleanup steps to still run and normal error printing to still happen and buying into the error recorder seemed like the best way to do that. But in taking another look I noticed a bug in my own code, where PreRun hooks didn't know to skip if there was already an error. Fixed in [83a7868](https://github.com/broadinstitute/thelma/pull/282/commits/83a78688f541ec9522d3863873925c95e0525a7d)